### ISSUE TITLE
[v4] | onAnimate gets invoked with toIndex: -1 when BottomSheetTextInput captures focus

### ISSUE THREAD

**Issue by udeyrishi** [gorhom/react-native-scroll-bottom-sheet#1441]
# Bug

The `onAnimate` callback provided to `BottomSheet` gets invoked with `toIndex === -1` if the `BottomSheetTextInput` captures focus. Without the use of `BottomSheetTextInput`, `onAnimate(-1)` usually signals that the sheet is animating away, so this behaviour seems odd. I don't see any related notes in the [docs](https://gorhom.github.io/react-native-bottom-sheet/keyboard-handling/) either, but apologies if I'm using the component incorrectly.

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet            | 4.4.7   |
| react-native                    | 0.71.8   |
| react-native-reanimated         | 3.3.0   |
| react-native-gesture-handler    | 2.12.0   |

## Steps To Reproduce

1. Use the `BottomSheet` component with the `onAnimate` prop set on it. `onAnimate` could simply be logging to console
2. Configure the body of `BottomSheet` to either contain a `BottomSheetTextInput`, or custom text input configured as suggested in the "tip" in the [docs](https://gorhom.github.io/react-native-bottom-sheet/keyboard-handling/)
3. Open the bottom sheet via your app's nav path
4. Click in the text input to capture focus
5. Monitor the console; `-1` should be logged

Describe what you expected to happen:

I'm not 100% sure, but this -1 behaviour seems odd. The most reasonable option IMO would be that the component tries to find the snap point that minimizes the height change required for the bottom sheet, finds the corresponding index, and invoke it with that.

## Reproducible sample code

```tsx
const onAnimate = (fromIndex: number, toIndex: number) => {
  console.log({fromIndex, toIndex})
}

<BottomSheet
  snapPoints={['25%']}
  onAnimate={onAnimate}
>
  <BottomSheetTextInput/>
</BottomSheet>
```

**Comment by zRelux** [comment]
Did you find a fix?

**Comment by zRelux** [comment]
This seems to happen only on android, ios works fine for me

**Comment by udeyrishi** [comment]
@zRelux I refactored my use case to use `onClose` instead to unblock myself; still decided to report it, in case it trips other people up. I think a callout in the docs might be a satisfactory fix? Not sure.

> This seems to happen only on android, ios works fine for me

I noticed the issue on iOS, and I verified it via the small isolated sample that I shared above. I haven't actually tested the behaviour on Android.
